gadget-core.Dockerfile

```diff
@@ -18,13 +18,18 @@ RUN set -ex; \
 	export DEBIAN_FRONTEND=noninteractive; \
 	apt-get update && \
 	apt-get install -y gcc make ca-certificates git clang \
-		software-properties-common libseccomp-dev llvm && \
-	add-apt-repository -y ppa:tuxinvader/kernel-build-tools && \
+		software-properties-common libelf-dev pkg-config libseccomp-dev llvm && \
 	apt-add-repository -y ppa:longsleep/golang-backports && \
 	apt-get update && \
-	apt-get install -y libbpf-dev golang-1.17 && \
+	apt-get install -y golang-1.17 && \
 	ln -s /usr/lib/go-1.17/bin/go /bin/go
 
+# Install libbpf-dev 0.7.0 from source to be cross-platform.
+RUN git clone https://github.com/libbpf/libbpf.git && \
+	cd libbpf/src && \
+	git checkout v0.7.0 && \
+	make -j$(nproc) install
```

Comment:
Do you know how much time it takes compared to the previous installation with apt-get? If it is comparable, then ok. If not, we could do that only on non-amd64.


Suggestion to download less from the git repo:
```
RUN git clone --branch v0.7.0 --depth 1 https://github.com/libbpf/libbpf.git && \
make -j$(nproc) -C libbpf/src install
```

Ditto in gadget-default.Dockerfile

Reply:
It is almost the same time, that is to say 10 minutes.

> we could do that only on non-amd64.

I would really like to avoid having things done only for a given architecture in Dockerfile.
Dockerfile were sadly not designed to support things like this:

```Dockerfile
#if TARGET_ARCH == .... 
# ...
#endif
```

> Suggestion to download less from the git repo:

Added, thank you!